lib/workspace.js

```diff
@@ -217,20 +217,25 @@ module.exports = {
         // FIXME: Check we are in "uninitialized"
         let result = await sqldb.queryOneRowAsync(sql.select_workspace_paths, {workspace_id});
         const {course_path, qid} = result.rows[0];
+        result = await sqldb.queryOneRowAsync(sql.select_workspace_version, {workspace_id});
+        const { workspace_version } = result.rows[0];
         const localPath = `${course_path}/questions/${qid}/workspace`;
-        const s3Path = `${config.workspaceS3Bucket}/workspace-${workspace_id}`;
+
+        const s3Name = `workspace-${workspace_id}-${workspace_version}/current`;
+        const s3Path = `${config.workspaceS3Bucket}/${s3Name}`;
```

Comment:
Very minor nit, but I'd rather do:
```suggestion
        const s3Name = `workspace-${workspace_id}-${workspace_version}`;
        const s3Path = `${config.workspaceS3Bucket}/${s3Name}/current`;
```

This way we can avoid the `replace()` call below to make `s3ZipPath`. I agree that in this case the use of `replace()` is safe, but I'm scared to use `replace()` on path-like strings because we run the risk that the replacement word appears more than once in the string.

Reply:
Went ahead and changed this in `interface` as well (this `replace` scheme started there as a lazy way to propagate the `*/current` change from 1 spot).